src/Skeleton/Skeleton.component.tsx

```diff
@@ -1,128 +1,130 @@
         textOnly
             ? css`
                   word-break: break-all;
-
                   color: ${colorStart};
-                  animation: color-change 2s ease-in-out infinite;
+                  animation: ${colorKeyframe(colorStart, colorEnd)} 2s
+                      ease-in-out infinite;
               `
             : css`
                   pointer-events: none;
-                  background: ${th.color(colorStart)};
-                  animation: background-change 2s ease-in-out infinite;
+                  background: ${colorStart};
+                  animation: ${bgKeyframe(colorStart, colorEnd)} 2s ease-in-out
+                      infinite;
 
                   // hide all children
                   && * {
                       visibility: hidden;
                       opacity: 0;
                   }
               `}
-
-    ${({ colorStart, colorEnd }) => css`
-        @keyframes background-change {
-            0% {
-                background: ${th.color(colorStart)};
-            }
-            50% {
-                background: ${th.color(colorEnd)};
-            }
-            100% {
-                background: ${th.color(colorStart)};
-            }
-        }
-        @keyframes color-change {
-            0% {
-                color: ${colorStart};
-            }
-            50% {
-                color: ${colorEnd};
-            }
-            100% {
-                color: ${colorStart};
-            }
-        }
-    `}
 `;
 
 export const Skeleton = ({
     className,
     children,
+    display,
     textLength,
+    textChar = '▆',
+    // This is the conversion from block width to the average width of
+    // a normal distribution of letters. We came up with it by eyeballing it
+    // for the Avenir Next font. We should probably make it themeable in the future.
+    blockRatio = 1 / 2.15,
     loading = true,
-    colorStart = th('skeleton.colorStart'),
-    colorEnd = th('skeleton.colorEnd'),
+    colorStart,
+    colorEnd,
     ...props
 }: SkeletonProps): React.ReactElement<StyledSkeletonProps> | any => {
     if (!loading) {
         return typeof children === 'undefined' ? null : children;
     }
 
-    // This is the conversion from block width to the average width of
-    // a normal distribution of letters. We came up with it by eyeballing it
-    // for the Avenir Next font. We should make it configurable via theme/prop
-    // or make further calculations to improve it.
-    const blockToAverageCharQuotient = 2.15;
+    const theme = React.useContext(ThemeContext);
+    const { colorStart: themeStart, colorEnd: themeEnd } = colorVariant({
+        ...props,
+        theme,
+    });
 
-    // This is the unicode block character that put together looks like one
-    // long bar. There are shorter and taller characters that could be used
-    // or a fatter or thinner bar. Therefore the bar thickness chould be made
-    // a prop in the future, mapping to different block characters.
```

Comment:
I wouldn't lose all of this comment. I believe `blockRatio` is the response to the second part of the comment, but the first part, `This is the unicode block character that put together looks like one long bar.`, is still helpful. Maybe move this up to where `textChar` is being assigned the character.